### PR TITLE
Properly cleanup channels after streams have been removed

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@vogdb please review

**Comment by vogdb** [comment]
started

File: test/spec/janus/plugin/audio.js [hunk 1]
```diff
@@ -124,10 +125,9 @@ describe('Audio plugin', function() {
         };
       });
 
-      it('should set channel', function(done) {
+      it('should add channel', function(done) {
         executeTransactionCallback().finally(function() {
-          expect(plugin.channel).to.be.instanceOf(Channel);
-          expect(plugin.channel.name).to.be.equal('channel-name');
+          expect(channels.add.withArgs('channel-name', 'channel-data').calledOnce);
```
Comment (vogdb) [inline test/spec/janus/plugin/audio.js]: Are you sure that it is correct? Channels.add has single-arg signature
```js
Channels.prototype.add = function(channel) {
```

File: test/spec/janus/plugin/video.js [hunk 2]
```diff
@@ -123,10 +124,9 @@ describe('Video plugin', function() {
         });
       });
 
-      it('should set channel', function(done) {
+      it('should add channel', function(done) {
         executeTransactionCallback().finally(function() {
-          expect(plugin.channel).to.be.instanceOf(Channel);
-          expect(plugin.channel.name).to.be.equal('channel-name');
+          expect(channels.add.withArgs('channel-name', 'channel-data').calledOnce);
```
Comment (vogdb) [inline test/spec/janus/plugin/video.js]: the same with the single-arg signature here

**Comment by vogdb** [comment]
the review is done

**Comment by tomaszdurka** [comment]
@vogdb please re-review

**Comment by vogdb** [comment]
lgtm

**Pull request merged**